### ISSUE TITLE
Slot bonus should apply to more than basic and mine income

### ISSUE THREAD

**Issue by stevietv** [Updownquark/OGame-ROI#4]
Slot bonus is currently only calculated over the Basic Income and the Mine Income.
However this bonus should be applied on total production, this including also:
* crawler production
* Plasma bonus
* collector bonus

Evidence:

![image](https://user-images.githubusercontent.com/1117625/94317232-246c4f80-ff86-11ea-8820-0802d0a57926.png)

![image](https://user-images.githubusercontent.com/1117625/94317250-2df5b780-ff86-11ea-898c-a3f275c7f74a.png)

Difference between metal production is 68k per hour.
17% * (Crawler + Plasma + Collector) = 17% * 398k = 68k

**Comment by Updownquark** [comment]
This is hopefully fixed in v0.9.4.  Please test if I missed anything.

The only things the slot bonus should *not* apply to now are officers and items.  Let me know if you think I'm wrong about either or both of those as well.

**Comment by stevietv** [comment]
seems to work for me now too.
For good measure after correlating app with game I added a 10% metal booster. This then only applies 10% to metal mine production only, and doesn't get boosted by other benefits. so your assumption about items is correct. not sure about officers though.
btw - how are items added in the app? think that you mentioned its not in the UI as yet?

**Comment by Updownquark** [comment]
Yeah, that's right. They're accounted for in the production code, but I've
never been able to decide where they should go in the UI. Suggestions are
welcome.

**Comment by stevietv** [comment]
I feel like the mines tab would be the best place? three columns for M
Boost C Boost and D Boost?

**Comment by Updownquark** [comment]
Added bonus items.  I think I can close this now.

**Comment by stevietv** [comment]
sorry to re-open this, but it seems slot bonus is applied to items.
In app a 10% booster shows 48302 per hour
In game a 10% booster shows 56513 per hour
discrepancy is 17% (i.e. slot bonus for slot 6)

**Comment by Updownquark** [comment]
Can you post me a screenshot of your in-game vs in-app resource settings?

**Comment by stevietv** [comment]
In Game:
![image](https://user-images.githubusercontent.com/1117625/94461475-7b5a6a80-01ba-11eb-86ee-252d3e0b8ac6.png)


In App

![image](https://user-images.githubusercontent.com/1117625/94461532-890ff000-01ba-11eb-8b2b-53518e756344.png)

As you can see, metal mine in game accounts first for slot bonus before the 10% booster is applied

**Comment by Updownquark** [comment]
I moved the slot update around a bit and applied it in a different way that I think better matches what the game engine is doing.  Please let me know if your numbers match or not.

**Comment by stevietv** [comment]
slot bonus seems great now. on both metal and crystal.
There does seem to be a minor mismatch on crawler and plasma production when a planet has a slot bonus.

1. Planet with metal slot bonus

![image](https://user-images.githubusercontent.com/1117625/94520333-60c0d980-022c-11eb-8adb-a8472e3ed105.png)

![image](https://user-images.githubusercontent.com/1117625/94520345-65858d80-022c-11eb-9d68-d36820678009.png)


2. Planet with crystal slot bonus

![image](https://user-images.githubusercontent.com/1117625/94520387-7afab780-022c-11eb-9260-4068776e80ac.png)
![image](https://user-images.githubusercontent.com/1117625/94520408-84841f80-022c-11eb-98fc-fd550b5d4231.png)

**Comment by Updownquark** [comment]
Alright, this should be fixed.  Try it out with the latest, please

**Comment by stevietv** [comment]
on my planet used in example 1, they are now only 1 production off on each line:

![image](https://user-images.githubusercontent.com/1117625/94520333-60c0d980-022c-11eb-8adb-a8472e3ed105.png)
![image](https://user-images.githubusercontent.com/1117625/94651622-0c376000-02f9-11eb-9577-bbb5fb2ece20.png)

however, I would say that your calculator is correct, and the rounding is an error on the display of gameforge.
Given that it I add all metal up it comes to 1,085,577 or 1,086M, which gameforge presents as 1,085M (they round down and not off)

**Comment by Updownquark** [comment]
That annoys me very much, but it points to something I'm doing differently and OGame that will be just slightly harder to fix.  I was assuming they're using integers for hourly production, so I am too.  It appears they're using floating point values for accumulation and just rounding them down in the production display.

One day I will do this as well, because it bothers me to be off at all.  But there are higher priorities.
